CHANGELOG.md

```diff
@@ -98,6 +98,11 @@ Adding a new version? You'll need three changes:
   method is adopted to keep the compatibility with traditional router on
   maximum effort.
   [#4240](https://github.com/Kong/kubernetes-ingress-controller/pull/4240)
+- When a translated Kong configuration is empty in DB-less mode, the controller
+  will now send the configuration with a single empty `Upstream`. This is to make
+  Gateways using `/status/ready` as their health check ready after receiving the
+  initial configuration (even if it's empty).
```

Comment:
Do you mind reminding me what's the reason for this? I do #3499 but as I see it now is that this will circumvent the `/status/ready` functionality: i.e. instead of waiting for first non-empty config we'll send the stub to make the GW ready which will serve basically nothing.

Reply:
It's to make sure a Gateway always gets ready after getting the first (empty or not) config update from KIC. Effectively, Gateways using `/status/ready` will stay not-ready until KIC sends the first config update to them. IMO it's expected to have KIC configure Gateways with an empty config when there's nothing in K8s API to translate, and assuming these Gateways will be ready to serve traffic in such case, isn't it?

This will also ensure that a typical scenario we run in tutorials etc. is not broken:
1. Deploy KIC + Kong using helm chart. 
2. `GET /path` returns 404.

Without this workaround, we'd have no Kong instances ready to serve and respond 404s.

Reply:
👍